lib/ast-utils.js

```diff
@@ -680,5 +680,46 @@ module.exports = {
         }
 
         return null;
+    },
+
+    /**
+     * Get directives from directive prologue of a Program or Function node.
+     * @param {ASTNode} node - The node to check.
+     * @returns {ASTNode[]|null} The directives found in the directive prologue or
+     * null if none are found.
+     */
+    getDirectivePrologue(node) {
+        if (
+            node.type !== "Program" &&
+            node.type !== "FunctionDeclaration" &&
+            node.type !== "FunctionExpression" &&
+            node.type !== "ArrowFunctionExpression" ||
+
+            // Do not check arrow functions with implicit return.
+            // `() => "use strict";` returns the string `"use strict"`.
+            node.type === "ArrowFunctionExpression" && node.body.type !== "BlockStatement"
+        ) {
+            return null;
+        }
+
+        const statements = node.type === "Program" ? node.body : node.body.body;
+        const directives = [];
+
+        for (const statement of statements) {
+            if (
+                statement.type === "ExpressionStatement" &&
+                statement.expression.type === "Literal"
+            ) {
+                directives.push(statement);
+            } else {
+                if (!directives.length) {
+                    return null;
```

Comment:
Personally, I prefer an empty array instead of `null`.

(currently, this returns an empty array if the function is empty?)